Sources/WalletConnectIdentity/IdentityClient.swift

```diff
@@ -1,62 +1,88 @@
+        self.networkClient = networkClient
         self.logger = logger
     }
 
-    func register(account: Account, onSign: SigningCallback) async throws -> String {
+    public func register(account: Account, onSign: SigningCallback) async throws -> String {
         let pubKey = try await identityService.registerIdentity(account: account, onSign: onSign)
         logger.debug("Did register an account: \(account)")
         return pubKey
     }
 
-    func goPublic(account: Account) async throws {
+    public func goPublic(account: Account) async throws {
         let inviteKey = try await identityService.registerInvite(account: account)
         try await subscribeForInvites(inviteKey: inviteKey)
         logger.debug("Did goPublic an account: \(account)")
     }
 
-    func unregister(account: Account, onSign: SigningCallback) async throws {
+    public func unregister(account: Account, onSign: SigningCallback) async throws {
         try await identityService.unregister(account: account, onSign: onSign)
         logger.debug("Did unregister an account: \(account)")
     }
 
-    func goPrivate(account: Account) async throws {
+    public func goPrivate(account: Account) async throws {
         let inviteKey = try await identityService.goPrivate(account: account)
         unsubscribeFromInvites(inviteKey: inviteKey)
         logger.debug("Did goPrivate an account: \(account)")
     }
 
-    func resolve(account: Account) async throws -> String {
+    public func resolveInvite(account: Account) async throws -> String {
         return try await identityService.resolveInvite(account: account)
     }
+
+    public func resolveIdentity(iss: String) async throws -> Account {
+        return try await identityService.resolveIdentity(iss: iss)
+    }
+
+    public func signAndCreateWrapper<T: JWTClaimsCodable>(
+        payload: T,
+        account: Account
+    ) throws -> T.Wrapper {
+        let identityKey = try identityStorage.getIdentityKey(for: account)
+        return try payload.signAndCreateWrapper(keyPair: identityKey)
+    }
+
+    public func getInviteKey(for account: Account) throws -> AgreementPublicKey {
+        return try identityStorage.getInviteKey(for: account)
+    }
 }
 
-private extension RegistryService {
+private extension IdentityClient {
 
     func subscribeForInvites(inviteKey: AgreementPublicKey) async throws {
         let topic = inviteKey.rawRepresentation.sha256().toHexString()
         try kms.setPublicKey(publicKey: inviteKey, for: topic)
-        try await networkingInteractor.subscribe(topic: topic)
+        try await networkClient.subscribe(topic: topic)
     }
 
     func unsubscribeFromInvites(inviteKey: AgreementPublicKey) {
         let topic = inviteKey.rawRepresentation.sha256().toHexString()
         kms.deletePublicKey(for: topic)
-        networkingInteractor.unsubscribe(topic: topic)
+        networkClient.unsubscribe(topic: topic)
     }
 }
```

Comment:
could this be a part of the Chat sdk?
Identity Package for client->server interaction

but this part is just chat peer to peer